### ISSUE TITLE
Cannot use Dart declared tables inside *.drift file

### ISSUE THREAD

**Issue by ky1vstar** [simolus3/moor#2273]
I've migrated `drift` to the latest version and now I see a warning when executing a `build_runner build`.

<details>
  <summary>lib/data_layer/database/database.dart</summary>
  
```dart
import 'package:drift/drift.dart';
/* ... */

part 'database.g.dart';

@DriftDatabase(
  tables: [Devices, Shots, ShotsAdditionalInfo],
  include: {'queries.drift'},
)
class Database extends _$Database {
  /* ... */
  
  SingleSelectable<ShotTrackingInfo> shotTrackingInfo([
    Expression<bool> Function(Shots tbl)? filter,
  ]) {
    final predicate = filter?.call(shots) ?? const CustomExpression('(TRUE)');
    return _shotTrackingInfo(predicate, predicate, predicate, predicate);
  }
}

/* ... */

class Devices extends Table {
  /* ... */
}

class Shots extends Table {
  /* ... */
}

@DataClassName("ShotAdditionalInfo")
class ShotsAdditionalInfo extends Table {
  /* ... */
}
```
</details>

<details>
  <summary>lib/data_layer/database/queries.drift</summary>

```sql
import 'database.dart';

_shotTrackingInfo AS ShotTrackingInfo: WITH
    countPerDay(countPerDay) AS (
      SELECT
        COUNT(shots.id)
      FROM
        shots
      WHERE
        $predicate1
      GROUP BY
        DATE(shots.end_time, 'unixepoch')
    ),
    countPerWeek(countPerWeek) AS (
      SELECT
        COUNT(shots.id)
      FROM
        shots
      WHERE
        $predicate2
      GROUP BY
        strftime(
          '%Y-%W',
          DATE(shots.end_time, 'unixepoch')
        )
    ),
    countPerMonth(countPerMonth) AS (
      SELECT
        COUNT(shots.id)
      FROM
        shots
      WHERE
        $predicate3
      GROUP BY
        strftime(
          '%Y-%m',
          DATE(shots.end_time, 'unixepoch')
        )
    ),
    -- https://blog.jooq.org/how-to-find-the-longest-consecutive-series-of-events-in-sql/
    dates(date) AS (
      SELECT
        DISTINCT DATE(shots.end_time, 'unixepoch')
      FROM
        shots
      WHERE
        $predicate4
    ),
    groupsByDate AS (
      SELECT
        ROW_NUMBER() OVER (
          ORDER BY
            date
        ) AS rn,
        date(
          date,
          '-' || ROW_NUMBER() OVER (
            ORDER BY
              date
          ) || ' day'
        ) AS grp,
        date
      FROM
        dates
    ),
    latestConsecutiveDates AS (
      SELECT
        COUNT(*) AS consecutiveDates,
        MIN(date) AS minDate,
        MAX(date) AS maxDate
      FROM
        groupsByDate
      GROUP BY
        grp
      ORDER BY
        MAX(date) DESC
      LIMIT
        1
    ), maxConsecutiveDates AS (
      SELECT
        COUNT(*) AS consecutiveDates
      FROM
        groupsByDate
      GROUP BY
        grp
      ORDER BY
        COUNT(*) DESC
      LIMIT
        1
    )
    SELECT
      IFNULL(
        AVG(countPerDay),
        0
      ) AS "averagePerDay",
      IFNULL(
        AVG(countPerWeek),
        0
      ) AS "averagePerWeek",
      IFNULL(
        AVG(countPerMonth),
        0
      ) AS "averagePerMonth",
      CASE WHEN latestConsecutiveDates.maxDate BETWEEN DATE('now', '-1 day')
      AND DATE('now', '+1 day', '-0.001 second') THEN IFNULL(
        latestConsecutiveDates.consecutiveDates,
        0
      ) ELSE 0 END AS "currentDaysStreak",
      IFNULL(
        maxConsecutiveDates.consecutiveDates,
        0
      ) as "maximumDaysStreak"
    FROM
      countPerDay,
      countPerWeek,
      countPerMonth,
      latestConsecutiveDates,
      maxConsecutiveDates;
```
</details>

I've modified `toString()` method of `ResultSetAvailableInStatement` class of `sqlparser` package to see what is reported by `build_runner`

<details>
  <summary>Warning in build log</summary>

```
...
[WARNING] drift_dev on lib/data_layer/database/database.dart:
line 8, column 9: Could not find shots. Available are: TableReference: countPerDay, TableReference: countPerWeek, TableReference: countPerMonth, TableReference: latestConsecutiveDates, TableReference: maxConsecutiveDates
  ╷
8 │         shots
  │         ^^^^^
  ╵
... repeats several times
```
</details>

Version:
- `drift`: 2.4.2
- `drift_dev`: 2.4.1

P.S.
Also it looks like now SQL inside drift files is "minified" incorrectly:

<details>
  <summary>Runtime Exception</summary>

```
SqliteException(1): near "SELECTROW_NUMBER": syntax error, SQL logic error (code 1)
  Causing statement: WITH countPerDay (countPerDay) AS (SELECT COUNT(shots.id) FROM shots WHERE "device_id" = ?1 GROUP BY DATE(shots.end_time, 'unixepoch')), countPerWeek (countPerWeek) AS (SELECT COUNT(shots.id) FROM shots WHERE "device_id" = ?2 GROUP BY strftime('%Y-%W', DATE(shots.end_time, 'unixepoch'))), countPerMonth (countPerMonth) AS (SELECT COUNT(shots.id) FROM shots WHERE "device_id" = ?3 GROUP BY strftime('%Y-%m', DATE(shots.end_time, 'unixepoch'))), dates (date) AS (SELECT DISTINCT DATE(shots.end_time, 'unixepoch') FROM shots WHERE "device_id" = ?4), groupsByDate AS (SELECTROW_NUMBER()OVER (ORDER BY date RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW EXCLUDE NO OTHERS) AS rn, date(date, '-' ||ROW_NUMBER()OVER (ORDER BY date RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW EXCLUDE NO OTHERS) || ' day') AS grp, date FROM dates), latestConsecutiveDates AS (SELECT COUNT(*) AS consecutiveDates, MIN(date) AS minDate, MAX(date) AS maxDate FROM groupsByDate GROUP BY grp ORDER BY MAX(date) DESC LIMIT 1), maxConsecutiveDates AS (SELECT COUNT(*) AS consecutiveDates FROM groupsByDate GROUP BY grp ORDER BY COUNT(*) DESC LIMIT 1) SELECT IFNULL(AVG(countPerDay), 0) AS averagePerDay, IFNULL(AVG(countPerWeek), 0) AS averagePerWeek, IFNULL(AVG(countPerMonth), 0) AS averagePerMonth, CASE WHEN latestConsecutiveDates.maxDate BETWEEN DATE('now', '-1 day') AND DATE('now', '+1 day', '-0.001 second') THEN IFNULL(latestConsecutiveDates.consecutiveDates, 0) ELSE 0 END AS currentDaysStreak, IFNULL(maxConsecutiveDates.consecutiveDates, 0) AS maximumDaysStreak FROM countPerDay,countPerWeek,countPerMonth,latestConsecutiveDates,maxConsecutiveDates
```
</details>

P.P.S.
Previous version I've been using is 1.7.1. It worked with neither `build_runner` warning nor runtime exception 😥

**Comment by simolus3** [comment]
Thanks for the report! I think the formatting issue is actually responsible for the tables not being found as well - ideally, both should be fixed in 7175b83d2b9955a41a31575de24015e62fbb4704.

Can you add this to your `pubspec.yaml`, run `flutter pub run build_runner clean` and `flutter pub run build_runner build -d` and check whether this fixes the problem when using drift 2?

```yaml
dependency_overrides:
  drift_dev:
    git:
      url: https://github.com/simolus3/drift.git
      ref: develop
      path: drift_dev
  sqlparser:
    git:
      url: https://github.com/simolus3/drift.git
      ref: develop
      path: sqlparser
```

**Comment by ky1vstar** [comment]
@simolus3 thanks for blazingly fast reply. It fixes my issue.
How do you think when this fix gonna be released?

**Comment by simolus3** [comment]
Thanks for the confirmation! I've just published `sqlparser: ^0.26.1` which contains the fix.
